static/js/containers/DashboardPage.js

```diff
@@ -429,7 +430,11 @@ class DashboardPage extends React.Component {
 
   addCourseEnrollment = (courseId: string): Promise<*> => {
     const { dispatch } = this.props;
-    return dispatch(addCourseEnrollment(courseId)).catch(() => {
+    return dispatch(actions.courseEnrollments.post(courseId)).then(courseId => {
```

Comment:
There's a bug here, the argument `course_id` for the function passed to `then` is the same as the argument to `addCourseEnrollment` so it ends up passing a course object (instead of id) to `showEnrollPayLaterSuccessMessage` and the message doesn't show up (but doesn't error because the comparison is valid).